warehouse/internal/repo/source_test.go

```diff
@@ -0,0 +1,460 @@
+package repo_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/samber/lo"
+
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/repo"
+)
+
+//nolint:unparam
+func sourcesJobs(
```

Comment:
[optional] can we have test helpers at the bottom of the file? it's to increase readability, since the first thing one looks for are the tests themselves.